rust/cloud-storage/properties/src/outbound/properties_pg_repo.rs

```diff
@@ -132,4 +132,34 @@ impl PropertiesRepo for PropertiesPgRepo {
             },
         }
     }
+
+    #[tracing::instrument(skip(self))]
+    async fn get_entity_name(
+        &self,
+        entity_id: &str,
+        entity_type: EntityType,
+    ) -> Result<Option<String>, Self::Err> {
+        // Task and Document entities are both stored in the Document table
+        // (tasks are documents with sub_type='task')
+        match entity_type {
+            EntityType::Task | EntityType::Document => {
+                match macro_db_client::document::get_document_name(&self.pool, entity_id).await {
+                    Ok(name) => Ok(Some(name)),
+                    Err(e) => {
+                        // If document/task doesn't exist, return None instead of error
+                        if let Some(db_err) = e.downcast_ref::<sqlx::Error>()
+                            && matches!(db_err, sqlx::Error::RowNotFound)
+                        {
+                            return Ok(None);
+                        }
+                        Err(e)
+                    }
+                }
+            }
+            _ => {
+                // Other entity types not yet supported
+                Ok(None)
+            }
+        }
+    }
```

Comment:
I think it would be better to be explicit here. take `Either<DocumentId, TaskId>` and make it explicit to the caller.